src/providers/googleProvider.ts

```diff
@@ -1,59 +1,108 @@
 import AbstractProvider, {
   EndpointArgument,
-  LatLng,
   ParseArgument,
+  ProviderOptions,
+  SearchArgument,
   SearchResult,
 } from './provider';
+import { Loader } from '@googlemaps/js-api-loader';
 
-export interface RequestResult {
-  results: RawResult[];
-  status: string;
+// See https://developers.google.com/maps/documentation/javascript/geocoding
+// for insight into the GeocoderResult.
+interface RequestResult {
+  results: google.maps.GeocoderResult[];
 }
 
-export interface RawResult {
-  address_components: {
-    long_name: string;
-    short_name: string;
-    types: string[];
-  }[];
-  formatted_address: string;
-  geometry: {
-    location: LatLng;
-    location_type: string;
-    viewport: {
-      northeast: LatLng;
-      southwest: LatLng;
-    };
-  };
-  place_id: string;
-  plus_code: {
-    compound_code: string;
-    global_code: string;
-  };
-  types: string[];
+interface GeocodeError {
+  code: Exclude<google.maps.GeocoderStatus, google.maps.GeocoderStatus.OK>;
+  endpoint: 'GEOCODER_GEOCODE';
+  message: string;
+  name: 'MapsRequestError';
+  stack: string;
 }
 
+// TODO - given that we need to instantiate a GoogleProvider
+// with a geoCoder arg, what's the best way to actually deal with
+// this to make it as easy as possible for the end user?
+// We could perhaps instruct them to call loadGeocoder() and
+// pass the results as the second arg... or wrap it all in a promise
+// and return that instead?
 export default class GoogleProvider extends AbstractProvider<
   RequestResult,
-  RawResult
+  google.maps.GeocoderResult
 > {
-  searchUrl = 'https://maps.googleapis.com/maps/api/geocode/json';
+  geoCoder: google.maps.Geocoder;
 
-  endpoint({ query }: EndpointArgument) {
-    const params = typeof query === 'string' ? { address: query } : query;
-    return this.getUrl(this.searchUrl, params);
+  constructor(geoCoder: google.maps.Geocoder) {
+    super();
+    this.geoCoder = geoCoder;
```

Comment:
TODO - maybe we want to add some logic here to ensure that the `geoCoder` argument has been provided.

Reply:
Fixed in 6cb96060d3d9b619cf2b593925b9d6855a3d4cec